src/main/kotlin/com/github/treepat/target_tree/default_tree/DefaultTargetTreeNode.kt

```diff
@@ -11,20 +11,31 @@ class DefaultTargetTreeNode(
     override var children: List<TargetTreeNode> = mutableListOf()
 ) : TargetTreeNode, Comparable<DefaultTargetTreeNode> {
 
-    var parent: TargetTreeNode? = null
+    var posAsChild: Int = -1
+    var parent: DefaultTargetTreeNode? = null
 
-    override fun moveToRightSibling(): TargetTreeNode? = (parent as? DefaultTargetTreeNode)?.getRightSibling(this)
+    override fun moveToRightSibling(): TargetTreeNode? {
+        if (posAsChild != -1 && posAsChild + 1 < parent?.children!!.size) {
```

Comment:
We should avoid using `!!`, it can create NPE.

Reply:
This applies for all in the future.

Reply:
Well, my parent should always have children. I think it would be preferable getting this error as it means the tree construction is wrong.